docs/json_rpc_api.md

```diff
@@ -526,9 +526,8 @@ the method name for further details such as parameter and return information.
 |   |   |
 |---|---|
 |Method|setgenerate|
-|Parameters|1. `generate`: `(boolean, required)` set to `true` to enable generation, `false` to disable it.<br />2. `genproclimit`: `(numeric, optional)` the number of processors (cores) to limit generation to or `-1` for default.|
+|Parameters|1. `generate`: `(boolean, required)` set to `true` to enable generation, `false` to disable it.<br />2. `genproclimit`: `(numeric, optional)` the number of processors (cores) to limit generation to or `-1` for default.<br /> 3. `miningaddr`: `(string array, optional)` the list of mining addresses |
```

Comment:
I'd remove 'optional' from `genproclimit`, because it's no longer optional in this case. It's more like 'optional-if'

Reply:
I think this is clear and does not require more explanation. We have only positional arguments parser so arguments always form the "chain" of parameters and middle ones are optional only if further ones are omitted. You can check for example `searchrawtransactions` - it has 5 parameters and all except first are optional.